backend/src/test/java/de/benjaminranft/spotown/controller/DiscoveryControllerIntegrationTest.java

```diff
@@ -0,0 +1,169 @@
+package de.benjaminranft.spotown.controller;
+
+import de.benjaminranft.spotown.dao.UserDao;
+import de.benjaminranft.spotown.dto.AddDiscoveryDto;
+import de.benjaminranft.spotown.dto.LoginDto;
+import de.benjaminranft.spotown.model.Discovery;
+import de.benjaminranft.spotown.model.User;
+import de.benjaminranft.spotown.utils.IdUtils;
+import de.benjaminranft.spotown.utils.TimestampUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.*;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.test.context.TestPropertySource;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(properties = {"jwt.secretkey=somesecrettoken"})
+public class DiscoveryControllerIntegrationTest {
+
+    @LocalServerPort
+    private int port;
+
+    @MockBean
+    private IdUtils idUtils;
+
+    @MockBean
+    private TimestampUtils timestampUtils;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @Autowired
+    private UserDao userDao;
+
+    @BeforeEach
+    public void setUpDb() {
+        Instant timestamp = Instant.parse("2020-11-18T18:35:24.00Z");
+        String passwordBenjamin = new BCryptPasswordEncoder().encode("password");
+        String passwordHeinz = "$2a$10$0HZGmicEH786L.HeSIjhOuvIK3ixlYij4luVHBNAUtXqKus79t/FS";
```

Comment:
Better use encode("password")

Reply:
Better use different passwords for both user

Reply:
> Better use encode("password")

I couldn't figure out how to make it work that way, because the encoding always returned varying results